MonitorControl/UI/SliderHandler.swift

```diff
@@ -45,13 +46,130 @@ class SliderHandler {
+      item.view = view
+      menu.insertItem(item, at: 0)
+    } else {
+      slider.frame.size.width = 180
+      slider.frame.origin = NSPoint(x: 15, y: 5)
+      let view = NSView(frame: NSRect(x: 0, y: 0, width: slider.frame.width + 30, height: slider.frame.height + 10))
+      let sliderHeaderItem = NSMenuItem()
+      let attrs: [NSAttributedString.Key: Any] = [.foregroundColor: NSColor.systemGray, .font: NSFont.systemFont(ofSize: 12)]
+      sliderHeaderItem.attributedTitle = NSAttributedString(string: title, attributes: attrs)
+      view.addSubview(slider)
+      item.view = view
+      menu.insertItem(item, at: 0)
+      menu.insertItem(sliderHeaderItem, at: 0)
+    }
+
+    var values: (UInt16, UInt16)?
+    let delay = display.needsLongerDelay ? UInt64(40 * kMillisecondScale) : nil
+
+    let tries = UInt(display.getPollingCount())
+    os_log("Polling %{public}@ times", type: .info, String(tries))
+
+    var (currentValue, maxValue) = (UInt16(0), UInt16(0))
+
+    if display.isSw(), command == Utils.Command.brightness {
+      (currentValue, maxValue) = (UInt16(display.getSwBrightnessPrefValue()), UInt16(display.getSwMaxBrightness()))
+    } else {
+      if tries != 0, !(app.safeMode) {
+        values = display.readDDCValues(for: command, tries: tries, minReplyDelay: delay)
+      }
+      (currentValue, maxValue) = values ?? (UInt16(display.getValueExists(for: command) ? display.getValue(for: command) : 75), 100) // We set 100 as max value if we could not read DDC, the previous setting as current value or 75 if not present.
+    }
+    display.saveMaxValue(Int(maxValue), for: command)
+    display.saveValue(min(Int(currentValue), display.getMaxValue(for: command)), for: command) // We won't allow currrent value to be higher than the max. value
+    os_log("%{public}@ (%{public}@):", type: .info, display.name, String(reflecting: command))
+    os_log(" - current value: %{public}@ - from display? %{public}@", type: .info, String(currentValue), String(values != nil))
+    os_log(" - maximum value: %{public}@ - from display? %{public}@", type: .info, String(display.getMaxValue(for: command)), String(values != nil))
+
+    if command == .brightness {
+      if !display.isSw(), prefs.bool(forKey: Utils.PrefKeys.lowerSwAfterBrightness.rawValue) {
+        slider.maxValue = Double(display.getMaxValue(for: command) * 2)
+        slider.integerValue = Int(slider.maxValue) / 2 + Int(currentValue)
+      } else {
+        slider.integerValue = Int(currentValue)
+        slider.maxValue = Double(display.getMaxValue(for: command))
+      }
+    } else if command == .audioSpeakerVolume {
+      // If we're looking at the audio speaker volume, also retrieve the values for the mute command
+      var muteValues: (current: UInt16, max: UInt16)?
+
+      os_log("Polling %{public}@ times", type: .info, String(tries))
+      os_log("%{public}@ (%{public}@):", type: .info, display.name, String(reflecting: Utils.Command.audioMuteScreenBlank))
```

Comment:
Same for these logs? (move them to correct condition)

Reply:
You are right, I did change this one (commited, pushed).